File: jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXLabeled.php
<?php
namespace php\gui;
use php\gui\paint\UXColor;
use php\gui\text\UXFont;

/**
 * Class UXLabeled
 * @package php\gui
 */
abstract class UXLabeled extends UXControl
{
    /**
     * @var string
     */
    public $alignment = 'BASELINE_CENTER';

    /**
     * @var string
     */
    public $textAlignment = 'LEFT';

    /**
     * @var bool
     */
    public $wrapText = false;

    /**
     * @var bool
     */
    public $underline = false;

    /**
     * @var string
     */
    public $text;

    /**
     * @var UXFont
     */
    public $font;

    /**
     * @var UXNode
     */
    public $graphic;

    /**
     * @var double
     */
    public $graphicTextGap = 4;

    /**
     * @var UXColor
     */
    public $textColor;

    /**
     * @var string
     */
    public $ellipsisString = '...';

    /**
     * @var string LEFT, TOP, RIGHT, BOTTOM, TEXT_ONLY, GRAPHIC_ONLY
     */
    public $contentDisplay = 'LEFT';

    /**
     * @var bool
     */
    public $mnemonicParsing = false;
}

File: jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXLabeled.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.geometry.Pos;
import javafx.scene.Node;
import javafx.scene.control.ContentDisplay;
import javafx.scene.control.Labeled;
import javafx.scene.paint.Color;
import javafx.scene.paint.Paint;
import javafx.scene.text.Font;
import javafx.scene.text.TextAlignment;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.*;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Abstract
@Reflection.Name(JavaFXExtension.NS + "UXLabeled")
public class UXLabeled<T extends Labeled> extends UXControl<Labeled> {
    interface WrappedInterface {
        @Property
        Pos alignment();

        @Property
        ContentDisplay contentDisplay();

        @Property
        String ellipsisString();

        @Property
        double graphicTextGap();

        @Property
        String text();

        @Property
        TextAlignment textAlignment();

        @Property
        boolean underline();

        @Property
        boolean wrapText();

        @Property @Nullable
        Font font();

        @Property
        boolean mnemonicParsing();
    }

    public UXLabeled(Environment env, T wrappedObject) {
        super(env, wrappedObject);
    }

    public UXLabeled(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    @SuppressWarnings("unchecked")
    public T getWrappedObject() {
        return (T) super.getWrappedObject();
    }

    @Setter
    public void setGraphic(@Nullable Node node) {
        getWrappedObject().setGraphic(node);
    }

    @Getter
    public Node getGraphic() {
        return getWrappedObject().getGraphic();
    }

    @Setter
    public void setTextColor(Color color) {
        getWrappedObject().setTextFill(color);
    }

    @Getter
    public Color getTextColor() {
        Paint textFill = getWrappedObject().getTextFill();

        if (textFill instanceof Color) {
            return (Color) textFill;
        }

        return null;
    }
}
